### ISSUE TITLE
beacon 문의합니다.

### ISSUE THREAD

**Issue by SongHyunHo** [wizTurn/Android-SDK#65]
제가 아직 학생인데 beacon에 대해 모르는게 너무 많아서 좀 공부를 하는 중인데요

sdk를 돌려보면, 비콘의 uuid, major,minor, rx, tx....가 받아오는데에 대한 질문 2가지만 하겠습니다 ㅠ 
꼭 답변 부탁드립니다.
1. 비콘을 사용하려면 백앤드서버를 구축해야 한다고 알고있는데요, 백앤드 서버를 
   구축했다고 가정하고 비콘 서비스를 설명하면
   
   비콘은 계속 브로드캐스팅되고 있습니다.
   블루투스를 켠 상태의 스마트폰(비콘앱이 백그라운드에서 실행 중)이
   비콘의 브로드캐스팅 범위안에 들어오면  비콘의 무엇과 스마트폰의 주소를 서버에 전송 한 후 특정 
   이벤트를 다시 스마트폰의 주소로 보내서 서비스하는 걸로 미약하게나마 알고있는데요
   
   질문 1. 여기서 서버에 보내는 (스마트폰이 받아서 처리하는) 비콘의 고유 ID는 UUID(Universally   Unique Identifier인 것인가요 아니면 pebBLE(위즈턴비콘)을 보면 밑에 적힌, BD Address 인것인가요 ? 
   
   인터넷에서는 UUID라고 하는 사람도 있고 애매모호하게 비콘 ID라고 한 사람도 있어서 확실하게 하기위해 질문드립니다 ㅠㅠ
   
   질문 2. 이제 서버에서 비콘ID와 스마트폰의 주소를 받아서 서버에 등록된 비콘 ID를 확인한 후, 스마트폰에 해당 비콘이 등록된 이벤트를 보내는 것으로 알고있는데, 그렇다면 여기서 메시지를 받을 스마트폰의 주소? 이것이 무엇을 의미하는 것인가요 ? 이것도 UUID를 의미하는 것인가요? 아니면 다른 무언가를 의미하는 것인가요 ? 별도의 IP가 없이 처리되는 것 같아 IP기능을 하는 무언가가 있나 궁금합니다.

**Comment by jayjay-dio** [comment]
안녕하세요. 디오의 안드로이드 개발 담당자입니다.

비콘을 사용하기 위해 서버를 구축할 필요가 없습니다. 서비스의 시나리오 중에 비콘의 실시간 정보나 특정 정보를 원격 서버에 저장하거나 원격 서버에서 처리가 필요하다면 서버를 시스템에 포함시키는 것입니다. 단지 특정 비콘의 신호를 감지했을 때, 팝업만 띄우고 싶으시다면 스마트폰만 있으시면 됩니다.

**Comment by SongHyunHo** [comment]
안녕하세요 답변 정말 감사합니다.

제가 지금 하려는 것은 일종의 매장처럼 구성을 하고 싶었던 것입니다. 
먼저 고객의 핸드폰에 특정 매장의 비콘 앱이 설치되어 백그라운드에 실행 중에 있다는 가정을 합니다.

저는 비콘이 설치된 매장이라면, 매장의 별도의 서버에서 비콘의 ID에 따른 이벤트를 가지고 있고 

비콘은 계속 매장 주변의 고객에게 브로드캐스팅을 하고, 그 정보를 백그라운드에서 실행중인 앱이 받으면, 그 앱을 통하여 다시 알림을 주는 줄 알았는데 잘 못 알고 있었던 것인가요 ?

그럼, 비콘 서비스란, 별도의 백앤드 서버가 존재하지 않고 비콘아이디를 수신하는 특정 앱에서만 단독으로 처리되는 것인지 궁금합니다. 예를들어 특정매장의 세일쿠폰 같은 정보를 수신하는 것을 말합니다.

**Comment by jayjay-dio** [comment]
디오에서 제공하는 pebBLE 비콘, SDK, 샘플앱으로 비콘을 활용한 어떤 서비스를 할 것인지는 고객님의 몫입니다.
서비스 시나리오상 단말과 서버간의 통신이 필요하면 서버를 시스템에 포함시켜야할 것으로 보이고, 필요하지 않는 시나리오라면 서버를 시스템에 포함시키지 않아도 될 것입니다. 이는 개발하시려는 서비스 요구사항을 면밀히 분석하신 후 결정되어야할 문제로서 저희 디오의 pebBLE 비콘과는 큰 연관성이 없습니다.

**Comment by SongHyunHo** [comment]
그럼 비콘에서 보내는 신호는 UUID,major,minor가 맞는 건가요 

기본적인 비콘의 사용방법은 
비콘이 보내는 신호를 감지하고 싶으면
비콘의  UUID를 안드로이드에 넣어 감지되게하는 것이 맞는건가요?
더 세부적으로 감지하게 하려면(사용자 위치를 자세히) major,minor까지 넣고 이 부분이 햇갈리네요.

**Comment by jayjay-dio** [comment]
iBeacon BLE 비콘의 브로드캐스트하는 데이터에는 UUID, major, minor, RSSI, bd address(or mac address)가 포함되어 있습니다.

**Comment by SongHyunHo** [comment]
그러면, 안드로이드로 비콘신호를 감지한다고 가정했을때, 
UUID나 bd address를 사용한다고 했을때 예를들면 if문으로  if(UUID==value)처럼 감지하여도 제가 원하는 서비스를 할 수 있다는 것인가요?

**Comment by jayjay-dio** [comment]
네, 맞습니다. uuid, major, minor 를 사용하셔서 비콘을 판별해도 되고 bd address로 비콘을 판별해도 됩니다.

**Comment by SongHyunHo** [comment]
아 친절한 답변, 정말 감사합니다.정말 많은 도움이 되었어요 !
